tests/network/checkup_framework/utils.py

```diff
@@ -256,13 +256,15 @@ def wait_for_job_failure(job):
         job_status = TimeoutSampler(
             wait_timeout=timeout,
             sleep=1,
-            func=lambda: job.instance.status.conditions[0],
+            func=lambda: job.instance.status.conditions,
```

Comment:
filter(lambda condition: condition["status"] == Resource.Condition.Status.TRUE, job.instance.status.conditions)

Reply:
I don't understand what is the meaning of this suggestion.
Anyway, I tried implementing it, but it fails with
```
    func=filter(lambda condition: condition["status"] == Resource.Condition.Status.TRUE,
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```
IIUIC, it's the `condition` var that is not known.
@dbasunag Can you please explain what is the meaning of your suggestion, and why is it preferred over the current code? Thank you

Reply:
@yossisegev , using [filter](https://docs.python.org/3/library/functions.html#filter) you can get only the interesting parts reducing the amount of checks on the sample afterwards.
(you are left checking just the "type")

The code us suppose to look like this:
```python
func=lambda: filter(lambda cond: cond["status"] == job.Condition.Status.TRUE , job.instance.status.conditions),
```

It's just nicer.

For me it's not a must.

Reply:
Ack, thank you @EdDev